### ISSUE TITLE
Radio Buttons and Checkboxes with DPI setting

### ISSUE THREAD

**Issue by macraigh** [wkhtmltopdf/wkhtmltopdf#4216]
wkhtmltopdf version: 0.12.5.0
os: Windows 10 Pro

I am observing certain behaviors when using wkhtmltopdf. I am attempting to produce a pdf while using the --dpi 200 option. When I do, I notice that this changes the size of the checkboxes and radio buttons.
To observe this, take the following html

    <!DOCTYPE html>
    <html>
    <body>
        <h1>Heading</h1>
        <p>Paragraph</p>
        <input type="checkbox" />
        <label>Checkbox</label>
        <br>
        <input type="radio" />
        <label>Radio</label>
    </body>
    </html>

Comparing the produced pdf for this with the following two commands, you will notice these elements are much smaller when setting the dpi.
wkhtmltopdf <url> <path>
wkhtmltopdf --dpi 200 <url> <path>

For other reasons, I need the dpi set to 200. I have been attempting to work around this problem to make the checkboxes and radio buttons appear larger, but cannot find a solution. Even after using css to transform the elements, or specify their size explicitly, I find wkhtmltopdf resizes the elements back down. A couple of examples of these attempts are as follows

    <!DOCTYPE html>
    <html>
    <body>
        <h1>Heading</h1>
        <p>Paragraph</p>
        <br />
        <div style="height: 50px; width: 50px;">
            <input style="height: 100%; width: 100%;" type="checkbox" checked />
        </div>
        <label>Checkbox</label>
        <br />
        <div style="height: 50px; width: 50px;">
            <input style="height: 100%; width: 100%;" type="radio" checked />
        </div>
        <label>Radio</label>
    </body>
    </html>

and

    <!DOCTYPE html>
    <html>
    <body>
        <h1>Heading</h1>
        <p>Paragraph</p>
        <br />
        <input style="transform: scale(2);" type="checkbox" checked />
        <label>Checkbox</label>
        <br />
        <input style="transform: scale(2);" type="radio" checked />
        <label>Radio</label>
    </body>
    </html>

In summary, I would like to produce larger checkboxes/radio buttons while using --dpi 200. Does anyone have any suggestions on how to approach this?

**Comment by macraigh** [comment]
@PhilterPaper Thank you very much for the suggestion. I believe you are suggesting some styling along the lines of below? I tested a few different approaches to this. Understanding that this doesn't actually work out correctly when rendered in the browser, I have tested this with the nuances of wkhtmltopdf. Unfortunately it results in the same issue of the checkboxes/radio buttons being scaled down by the dpi setting.

    <!DOCTYPE html>
    <html>
    <body>
        <h1>Heading</h1>
        <p>Paragraph</p>
        <br />
        <input type="checkbox" style="font-size: 72pt;" checked />
        <label>Checkbox</label>
        <br />
        <div style="font-size: 72pt;">
            <input type="checkbox" checked />
            <label>Checkbox</label>
        </div>
        <br />
        <br />
        <input type="checkbox" style="font-size: 300%;" checked />
        <label>Checkbox</label>
        <br />
        <div style="font-size: 300%;">
            <input type="checkbox" checked />
            <label>Checkbox</label>
        </div>
    </body>
    </html>

**Comment by PhilterPaper** [comment]
Can you make a small (minimal) test case, and then look inside the PDF to see how it's handling a checkbox? I think it will be text, but it would be good to confirm this before spending any more time going down this road. You may need to use a PDF utility to uncompress streams in the PDF so you can read them. If you haven't done so already, search this bug list for discussion on checkboxes, just in case someone has run into this before. It's possible that text-oriented CSS like you tried may need to be modified to work with hidden text such as checkboxes. That's all I can think of at this time.

**Comment by KaptenJon** [comment]
This is still an issue i believe.

**Comment by tielur** [comment]
I'm running into this experience now as well on version 1.4.0

**Comment by joeysr20det** [comment]
I'm also running into this same issue with the latest version as of this time 0.12.6. When setting the DPI as described, checkboxes and radio buttons are tiny.

**Comment by MattHoneycutt** [comment]
Did anyone find a good workaround to this issue?  We're seeing the same thing.  I've tried every workaround I can find, but nothing seems to increase the size of the checkboxes and radio buttons.  I've tried switching to SVGs, but the SVGs are scaled down, too, no matter what I try.

**Comment by joeysr20det** [comment]
Matt,

We were never able to get around this issue. It appears as though the checkboxes and radio buttons may be getting replaced with checkbox and radio button characters from a PDF font. 

While we were not able to get around this, we did apply special CSS when rendering to PDF that applies some top and left margin to at least center their character so it appears to be in the correct place. This was has been a huge pain for us not having the ability to correct or style these to match the original controls our users see when filling out electronic forms. Hopefully this can/will be addressed soon.

Thanks again for all of the work on this project! We don't often use any external libraries in our software, but overall wkhtmltopdf has been great!

**Comment by MAnasKhalid** [comment]
input[type=radio] { transform: scale(3)!important; -webkit-transform: scale(3)!important; }
 input[type=checkbox] { transform: scale(3)!important; -webkit-transform: scale(3)!important; }
